src/main/java/net/sf/jabref/gui/maintable/MainTable.java

```diff
@@ -499,11 +499,14 @@ public void setSelected(int row) {
     }
 
     public int findEntry(BibEntry entry) {
-        return model.getTableRows().indexOf(entry);
-    }
-
-    public int findLastEntry(BibEntry entry) {
-        return model.getTableRows().lastIndexOf(entry);
+        EventList<BibEntry> tableRows = model.getTableRows();
```

Comment:
I am not sure, but I think this could be replaced a combination of stream,filter  and findFirst 
sth like : t`ableRows.stream().filter(bibEntry->bibEntry.getId().equals(entry.getId()).findFirst()

And it has the advantage to return an optional which is **way** better to return a -1 value. Returning 1 comes from the old times of programming where everything >0  was indicating succes and negative things like -1 indicated errors.  (Unfortunately I just can't find a source explaining it, but it is not considered good practice returning -1 nowadays if no element is found or error happend)